agent/pkg/controllers/entries_controller.go

```diff
@@ -72,23 +73,46 @@ func SyncEntries(c *gin.Context) {
 		c.JSON(http.StatusBadRequest, err)
 		return
 	}
+
 	if err := validation.Validate(syncParams); err != nil {
 		c.JSON(http.StatusBadRequest, err)
 		return
 	}
+
 	if up9.GetAnalyzeInfo().IsAnalyzing {
 		c.String(http.StatusBadRequest, "Cannot analyze, mizu is already analyzing")
 		return
 	}
 
-	rlog.Infof("Sync entries - creating token. env %s\n", syncParams.Env)
-	token, err := up9.CreateAnonymousToken(syncParams.Env)
-	if err != nil {
-		c.String(http.StatusServiceUnavailable, "Cannot analyze, mizu is already analyzing")
+	var (
+		token, model string
+		guestMode    bool
+	)
+	if syncParams.Token == "" {
+		rlog.Infof("Sync entries - creating token. env %s\n", syncParams.Env)
+		guestToken, err := up9.CreateAnonymousToken(syncParams.Env)
+		if err != nil {
+			c.String(http.StatusServiceUnavailable, "Failed creating anonymous token")
+			return
+		}
+
+		token = guestToken.Token
+		model = guestToken.Model
+		guestMode = true
+	} else {
+		token = fmt.Sprintf("bearer %s", syncParams.Token)
+		model = syncParams.Workspace
+		guestMode = false
+	}
+
+	modelRegex, _ := regexp.Compile("[A-Za-z0-9][-A-Za-z0-9_.]*[A-Za-z0-9]+$")
+	if len(model) > 63 || !modelRegex.MatchString(model) {
+		c.String(http.StatusBadRequest, "Invalid model name")
 		return
 	}
-	rlog.Infof("Sync entries - syncing. token: %s model: %s\n", token.Token, token.Model)
-	go up9.SyncEntriesImpl(token.Token, token.Model, syncParams.Env, syncParams.SleepIntervalSec)
+
+	rlog.Infof("Sync entries - syncing. token: %s model: %s\n", token, model)
+	go up9.SyncEntriesImpl(token, model, syncParams.Env, syncParams.SleepIntervalSec, guestMode)
```

Comment:
we can change it here as well to be aligned with the var names, something "upload interval'